src/main/java/com/google/api/codegen/transformer/ApiCallableTransformer.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    apiCallables.add(apiCallableBuilder.build());
+
+    if (methodConfig.isPageStreaming()) {
+      PageStreamingConfig pageStreaming = methodConfig.getPageStreaming();
+
+      ApiCallableView.Builder pagedApiCallableBuilder = ApiCallableView.newBuilder();
+      pagedApiCallableBuilder.type(ApiCallableType.PagedApiCallable);
+
+      String pagedResponseTypeName =
+          context
+              .getNamer()
+              .getAndSavePagedResponseTypeName(
+                  typeTable, pageStreaming.getResourcesField().getType());
+
+      pagedApiCallableBuilder.requestTypeName(
+          typeTable.getAndSaveNicknameFor(method.getInputType()));
+      pagedApiCallableBuilder.responseTypeName(pagedResponseTypeName);
+      pagedApiCallableBuilder.name(context.getNamer().getPagedCallableName(method));
+      pagedApiCallableBuilder.settingsFunctionName(
+          context.getNamer().getSettingsFunctionName(method));
+
+      apiCallables.add(pagedApiCallableBuilder.build());
+    }
+
+    return apiCallables;
+  }
+
+  private List<ApiCallSettingsView> generateApiCallableSettings(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    ModelTypeTable typeTable = context.getTypeTable();
+    Method method = context.getMethod();
+    MethodConfig methodConfig = context.getMethodConfig();
+
+    ApiCallSettingsView.Builder settings = ApiCallSettingsView.newBuilder();
+
+    settings.methodName(namer.getApiMethodName(method));
+    settings.requestTypeName(typeTable.getAndSaveNicknameFor(method.getInputType()));
+    settings.responseTypeName(typeTable.getAndSaveNicknameFor(method.getOutputType()));
+    settings.grpcTypeName(
+        typeTable.getAndSaveNicknameFor(namer.getGrpcContainerTypeName(context.getInterface())));
+    settings.grpcMethodConstant(namer.getGrpcMethodConstant(method));
+    settings.retryCodesName(methodConfig.getRetryCodesConfigName());
+    settings.retryParamsName(methodConfig.getRetrySettingsConfigName());
+
+    String notImplementedPrefix = "ApiCallableTransformer.generateApiCallableSettings - ";
+    settings.resourceTypeName(
+        namer.getNotImplementedString(notImplementedPrefix + "resourceTypeName"));
+    settings.pageStreamingDescriptorName(
+        namer.getNotImplementedString(notImplementedPrefix + "pageStreamingDescriptorName"));
+    settings.bundlingDescriptorName(
+        namer.getNotImplementedString(notImplementedPrefix + "bundlingDescriptorName"));
+
+    if (methodConfig.isPageStreaming()) {
+      namer.addPageStreamingCallSettingsImports(typeTable);
+      settings.type(ApiCallableType.PagedApiCallable);
+      TypeRef resourceType = methodConfig.getPageStreaming().getResourcesField().getType();
+      settings.resourceTypeName(typeTable.getAndSaveNicknameForElementType(resourceType));
+      settings.pageStreamingDescriptorName(namer.getPageStreamingDescriptorConstName(method));
+    } else {
+      if (methodConfig.isBundling()) {
```

Comment:
Given this context, it would make sense to use a normal else if. I think it ended up this way because in other places, there were some things shared by bundling & simple. I will change it here.